pkg/utils/utils.go

```diff
@@ -238,12 +240,27 @@ func HasDpdkDriver(pciAddr string) (bool, error) {
 
 // SaveNetConf takes in container ID, data dir and Pod interface name as string and a json encoded struct Conf
 // and save this Conf in data dir
-func SaveNetConf(cid, dataDir, podIfName string, conf interface{}) error {
-	netConfBytes, err := json.Marshal(conf)
+func SaveNetConf(cid, dataDir, podIfName string, netConf sriovtypes.NetConf) error {
+	netConfBytes, err := json.Marshal(netConf)
 	if err != nil {
 		return fmt.Errorf("error serializing delegate netconf: %v", err)
 	}
 
+	fixupObj := make(map[string]interface{})
```

Comment:
why use a new var called ``fixupObj`` when you could continue to use ``netConfBytes`` ?

Reply:
not sure I see a reason to Unmarshal json twice?

Reply:
ohh I misread, you marshal the netConf then unmarshal it afterwards

Reply:
apologies for the spam here - but I think my point still stands - if you just checked the DNS before marshalling, you would never have to marshal and unmarshal and then later marshal the fixed object again?

Reply:
we need this one to be from type [string]interface{} to be able and remove the DNS if needed

Reply:
ah understood okay, thanks for clarifying